integration-tests/smoke/ocr2_test.go

```diff
@@ -1,85 +1,241 @@
 package smoke_test
 
-//revive:disable:dot-imports
+// revive:disable:dot-imports
 import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
 	"math/big"
+	"os"
+	"time"
 
+	"github.com/dontpanicdao/caigo/gateway"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
-	it "github.com/smartcontractkit/chainlink-starknet/integration-tests"
+	"github.com/rs/zerolog/log"
+	"github.com/smartcontractkit/chainlink-relay/pkg/logger"
+	"github.com/smartcontractkit/chainlink/integration-tests/actions"
+	client "github.com/smartcontractkit/chainlink/integration-tests/client"
+
+	"github.com/smartcontractkit/chainlink-starknet/integration-tests/common"
 	"github.com/smartcontractkit/chainlink-starknet/relayer/ops"
-	"github.com/smartcontractkit/chainlink-testing-framework/actions"
-	"github.com/smartcontractkit/chainlink-testing-framework/blockchain"
-	"github.com/smartcontractkit/chainlink-testing-framework/client"
-	"github.com/smartcontractkit/chainlink-testing-framework/utils"
-	"github.com/smartcontractkit/helmenv/environment"
+	"github.com/smartcontractkit/chainlink-starknet/relayer/ops/devnet"
+	"github.com/smartcontractkit/chainlink-starknet/relayer/pkg/chainlink/ocr2"
+	"github.com/smartcontractkit/chainlink-starknet/relayer/pkg/starknet"
+)
+
+var (
+	keepAlive bool
 )
 
+func init() {
+	flag.BoolVar(&keepAlive, "keep-alive", false, "enable to keep the cluster alive")
+}
+
 var _ = Describe("StarkNET OCR suite @ocr", func() {
 	var (
-		err           error
-		nets          *blockchain.Networks
-		cls           []client.Chainlink
-		networkL1     blockchain.EVMClient
-		networkL2     blockchain.EVMClient
-		ocrDeployer   *it.OCRDeployer
-		starkDeployer *it.StarkNetContractDeployer
-		e             *environment.Environment
+		err                     error
+		linkTokenAddress        string
+		accessControllerAddress string
+		ocrAddress              string
+		sg                      *ops.StarknetGauntlet
+		t                       *common.Test
+		nAccounts               []string
+		serviceKeyL1            = "Hardhat"
+		serviceKeyL2            = "starknet-dev"
+		serviceKeyChainlink     = "chainlink"
+		chainName               = "starknet"
+		chainId                 = gateway.GOERLI_ID
```

Comment:
hmmm not sure if we want to share these, since eventually `common` would be used for `smoke` and `soak`
In my mind:
* `smoke`: quick test against local devnet
* `soak`: can be pointed to any network - devnet, testnet, mainnet